### PR TITLE
Clq service updates

### PR DESCRIPTION
**Pull request by rdingwell**
Changes required to work with CLQ service.   

The main changes are the addition of an apiKey to the Auth header and a modification to the way that outcomes are reported.   Previously the service was going to report the total number of patients surviving for a given treatment and timeframe.  This has changed to reporting a percentage of patients instead.   

There is also a change to deal with case sensitivity of the value of the tumor markers (Positive,Negative)  

## Submitter:

**Running the Application**

- [ ] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [ ] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/mcode-pilot/services/outcomes/CLQOutcomesService.js [hunk 1]
```diff
@@ -105,7 +107,7 @@ export default class CLQOutcomesService extends IOutcomesService {
                 code: code.code.code,
                 codeSystem: code.codeSystem.uri,
                 displayName: code.displayText.string,
-                value: value
+                value: _.startCase(_.toLower(value))
```
Comment (dehall) [inline src/mcode-pilot/services/outcomes/CLQOutcomesService.js]: Not a big deal but is the `toLower` call here necessary? From a glance at the docs it seems like  `startCase(toLower(x))` should === `startCase(x)`

Comment (rdingwell) [reply src/mcode-pilot/services/outcomes/CLQOutcomesService.js]: I'll take a look at the docs but I thought that startCase only capitalize the first letter and left the rest of the string alone.

Comment (rdingwell) [reply src/mcode-pilot/services/outcomes/CLQOutcomesService.js]: Some simple tests on strings and startCase will ensure that every word in the string is capitalized and leave the rest of the string alone.  So "HEY THAR" will stay "HEY THAR".

Comment (dehall) [reply src/mcode-pilot/services/outcomes/CLQOutcomesService.js]: Yeah I just tried it too. startCase is weird.

File: src/mcode-pilot/services/outcomes/CLQOutcomesService.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ export default class CLQOutcomesService extends IOutcomesService {
         super();
         this.serviceUrl = params.serviceUrl;
         this.timescale = params.timescale || [];
+        this.apiKey = params.apiKey;
+
```
Comment (Dtphelan1) [inline src/mcode-pilot/services/outcomes/CLQOutcomesService.js]:
```suggestion
```

**Comment by Dtphelan1** [comment]
1) Could `proportion_surviving` be changed to use lowerCamelCase. I can't immediately tell if we have control of that key-value specification or not.

**Comment by rdingwell** [comment]
@Dtphelan1 we do not have control over that key, it is coming from CLQ

**Comment by Dtphelan1** [comment]
Rebasing this branch as I finish review. Think this looks good - just want to check if there's explicitly anything I should be testing, or just giving it a visual lookover.

**Comment by Dtphelan1** [comment]
Looks like this isn't working when I reach out to the outcomes service. Is this expected? @rdingwell ? 
image

Is what happens with config: 
image

**Comment by Dtphelan1** [comment]
After testing with:
```
outcomes: { 
    "name" : "CLQOutcomesService",  
    "serviceUrl" : "http://moonshot-dev:5000/outcomes", 
    "timescale": ["1","3","5"] 
}
```

it works! Thanks Rob!

**Comment by Dtphelan1** [comment]
Rebased - waiting for tests to pass.

**Pull request merged**